### ISSUE TITLE
./pkgs/misc/vim-plugins/default.nix:2:1 called without required argument ‘Cocoa’, at ./lib/customisation.nix:56:12

### ISSUE THREAD

**Issue by steveeJ** [NixOS/nixpkgs#12931]
trace: https://gist.github.com/steveeJ/4ac6baacccbca09863f7

This happens for a while now. The following patch seems to help a little

```patch
diff --git a/pkgs/misc/vim-plugins/default.nix b/pkgs/misc/vim-plugins/default.nix
index 84386ef..74c6b64 100644
--- a/pkgs/misc/vim-plugins/default.nix
+++ b/pkgs/misc/vim-plugins/default.nix
@@ -2,7 +2,7 @@
 { fetchurl, bash, stdenv, python, go, cmake, vim, vimUtils, perl, ruby, unzip
 , which, fetchgit, fetchFromGitHub, fetchhg, fetchzip, llvmPackages, zip
 , vim_configurable, vimPlugins, xkb_switch, git
-, Cocoa
+, Cocoa ? null
 }:
 
 let
```
I get build failures of other packages later of which I don't know if they're related. Currently, bash is the next fail in line.

/cc @jagajaga

**Comment by jagajaga** [comment]
I've just built it myself from master branch and It worked for me... Can you post any more information?

**Comment by steveeJ** [comment]
Sure, whatever you need. I'm using a customized neovim, is that relevant?

**Comment by jagajaga** [comment]
Yes, please send it here.

-- 
Sincerely,
Arseniy Seroka

**Comment by steveeJ** [comment]
https://github.com/steveeJ/nix-expressions/blob/5848f88/configuration/common/pkg/neovim.nix

**Comment by jagajaga** [comment]
I suppose that's because of override function.. So let's apply your commit to avoid that in future.

**Comment by steveeJ** [comment]
This shouldn't be closed yet because the commit hasn't landed

**Comment by steveeJ** [comment]
Fixed in #13575.
